File: 2561-number-of-distinct-averages/2561-number-of-distinct-averages.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var distinctAverages = function(nums) {
    nums.sort((a,b) => a - b)
    avgContainer = new Set()

    while(nums.length > 1){
        big = nums.pop()
        small = nums.shift()
        avg = (big + small) / 2
        avgContainer.add(avg)
    }
    return avgContainer.size
};

File: 2561-number-of-distinct-averages/2561-number-of-distinct-averages.py
class Solution:
    def distinctAverages(self, nums: List[int]) -> int:
        nums.sort()
        avgContainer = set()

        while len(nums) > 1:
            big = nums.pop()
            small = nums.pop(0)

            avg = (big + small) / 2
            avgContainer.add(avg)
        return len(avgContainer)
        

        